torchdata/datapipes/iter/util/plain_text_reader.py

```diff
@@ -0,0 +1,237 @@
+        strip_newline: bool = True,
+        decode: bool = False,
+        encoding="utf-8",
+        errors: str = "ignore",
+        return_path: bool = True,
+    ):
+        self.source_datapipe = source_datapipe
+        self._helper = PlainTextReaderHelper(
+            skip_lines=skip_lines,
+            strip_newline=strip_newline,
+            decode=decode,
+            encoding=encoding,
+            errors=errors,
+            return_path=return_path,
+        )
+
+    def __iter__(self):
+        for path, stream in self.source_datapipe:
+            stream = self._helper.skip_lines(stream)
+            stream = self._helper.strip_newline(stream)
+            stream = self._helper.decode(stream)
+            stream = self._helper.return_path(stream, path=path)
+            yield from stream
+
+
+class _CSVBaseParserIterDataPipe(IterDataPipe):
+    def __init__(
+        self,
+        source_datapipe,
+        csv_reader,
+        *,
+        skip_lines: int = 0,
+        decode: bool = False,
+        encoding="utf-8",
+        errors: str = "ignore",
+        return_path: bool = True,
+        **fmtparams,
+    ):
+        self.source_datapipe = source_datapipe
+        self._csv_reader = csv_reader
+        self._helper = PlainTextReaderHelper(
+            skip_lines=skip_lines,
+            decode=decode,
+            encoding=encoding,
+            errors=errors,
+            return_path=return_path,
+        )
+        self.fmtparams = fmtparams
+
+    def __iter__(self):
+        for path, stream in self.source_datapipe:
+            stream = self._helper.skip_lines(stream)
+            stream = self._helper.decode(stream)
+            stream = self._csv_reader(stream, **self.fmtparams)
+            stream = self._helper.return_path(stream, path=path)
+            yield from stream
+
+
+@functional_datapipe("parse_csv")
+class CSVParserIterDataPipe(_CSVBaseParserIterDataPipe):
+    r"""
+    Iterable DataPipe that accepts a DataPipe consists of tuples of file name and CSV data stream.
+    This reads and returns the contents within the CSV files one row at a time (as a List
+    by default, depending on fmtparams).
+
+    Args:
+        source_datapipe: source DataPipe with tuples of file name and CSV data stream
+        skip_lines: number of lines to skip at the beginning of each file
+        strip_newline: if True, the new line character will be stripped
+        decode: if True, this will decode the contents of the file based on the specified encoding
+        encoding: the character encoding of the files (default='utf-8')
+        errors: the error handling scheme used while decoding
+        return_path: if True, each line will return a tuple of path and contents, rather
+            than just the contents
+    """
+
+    def __init__(
+        self,
+        source_datapipe,
+        *,
+        skip_header=0,
+        decode=True,
+        encoding="utf-8",
+        errors="ignore",
+        keep_filename=False,
```

Comment:
We now have two different arguments `keep_filename` and `return_path`. Could we use a same name for it?

Reply:
Sure. I would prefer `return_path`, because we actually return a path rather than just a file name.